bigml/api_handlers/sourcehandler.py

```diff
@@ -573,18 +573,20 @@ def update_composite_annotations(self, source, images_file,
                 offset * MAX_CHANGES: (offset + 1) * MAX_CHANGES]
             if new_batch:
                 source = self.update_source(source,
-                                            {"row_values": new_batch})
-                if source["error"] is not None:
+                                            {"row_values": new_batch})+
+                counter = 0
+                while source["error"] is not None and counter < MAX_RETRIES:
                     # retrying in case update is temporarily unavailable
-                    time.sleep(1)
+                    counter += 1
+                    time.sleep(counter)
                     source = self.get_source(source)
                     self.ok(source)
                     source = self.update_source(source,
                                                 {"row_values": new_batch})
-                    if source["error"] is not None:
-                        LOGGER.error("WARNING: Some annotations were not"
-                                     " updated (%s)",
-                                      new_batch)
+                if source["error"] is not None:
+                    LOGGER.error("WARNING: Some annotations were not"
+                                 " updated (%s)",
+                                  new_batch)
```

Comment:
perhaps it'd be a good idea to log also `source["error"]` here?

Reply:
Ok, though the error message and code are not very reprentative so far